src/components/installer/installer.ts

```diff
@@ -0,0 +1,187 @@
+'use strict';
+
+import * as download from 'download';
+import * as fs from 'fs';
+import mkdirp = require('mkdirp');
+import * as path from 'path';
+import * as tar from 'tar';
+import * as tmp from 'tmp';
+import * as vscode from 'vscode';
+import { Shell, Platform } from '../../shell';
+import { Errorable } from '../../wizard';
+import { exec } from 'child_process';
+
+export async function installKubectl(shell: Shell) : Promise<Errorable<void>> {
+    const tool = 'kubectl';
+    const installFolder = getInstallFolder(shell, tool);
+    mkdirp.sync(installFolder);
+    switch (shell.platform()) {
+        case Platform.Windows:
+        {
+            // Instruction say always to use this link for Windows, not to check the version first
+            const kubectlUrl = 'https://storage.googleapis.com/kubernetes-release/release/v1.9.6/bin/windows/amd64/kubectl.exe';
+            const downloadFile = path.join(installFolder, 'kubectl.exe');
+            const downloadResult = await downloadTo(kubectlUrl, downloadFile);
+            if (!downloadResult.succeeded) {
+                return { succeeded: false, result: null, error: [`Failed to download kubectl: ${downloadResult.error[0]}`] };
+            }
+            addPathToConfig(`vs-kubernetes.${tool}-path`, downloadFile);
+            return { succeeded: true, result: null, error: [] };
+        }
+        case Platform.Linux:
+        case Platform.MacOS:
+        {
+            const version = await getStableKubectlVersion();
+            if (!version.succeeded) {
+                return { succeeded: false, result: null, error: version.error };
+            }
+            const os = platformUrlString(shell.platform());
+            const kubectlUrl = `https://storage.googleapis.com/kubernetes-release/release/${version}/bin/${os}/amd64/kubectl`;
+            const downloadFile = path.join(installFolder, 'kubectl');
+            const downloadResult = await downloadTo(kubectlUrl, downloadFile);
+            if (!downloadResult.succeeded) {
+                return { succeeded: false, result: null, error: [`Failed to download kubectl: ${downloadResult.error[0]}`] };
+            }
+            fs.chmodSync(downloadFile, '0777');
+            addPathToConfig(`vs-kubernetes.${tool}-path`, downloadFile);
+            return { succeeded: true, result: null, error: [] };
+        }
+    }
+}
+
+async function getStableKubectlVersion() : Promise<Errorable<string>> {
+    const downloadResult = await downloadToTempFile('https://storage.googleapis.com/kubernetes-release/release/stable.txt');
```

Comment:
why not just make a http request against this to read the version?

Reply:
`<confession>` I was being lazy because I had already written the 'download to file' code and was struggling with the documentation to get the response body directly. `</confession>`

I've now ditched `request` in favour of `download` so will probably end up sticking with this approach though I agree it is entirely superfluous going via the filesystem!